### PR TITLE
Add consumer

### PR DESCRIPTION
**Pull request by GTLiSunnyi**

### CODE REVIEW

File: templates/consumer/go/app/app.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package app
+
+import (
+	"fmt"
+
+	"github.com/irisnet/service-gen/service"
+	"github.com/irisnet/service-gen/{{service_name}}"
+	"github.com/irisnet/service-gen/types"
+	servicesdk "github.com/irisnet/service-sdk-go/service"
+	log "github.com/sirupsen/logrus"
```
Comment (chengwenxi) [inline templates/consumer/go/app/app.go]: go format

File: templates/consumer/go/app/app.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+// Invoke providers' service
+func (app App) Invoke(invokeConfig servicesdk.InvokeServiceRequest) {
+	reqCtxID, reqID, err := app.ServiceClient.InvokeService(invokeConfig)
+	if err != nil {
+		fmt.Printf("failed to invoke service request listener, err: %s \n", err.Error())
+		return
+	}
+	fmt.Println("Successfully invoke service.")
+	fmt.Println("reqCtxID:", reqCtxID)
+	fmt.Println("reqID:", reqID)
```
Comment (chengwenxi) [inline templates/consumer/go/app/app.go]: subscribe response and print it.

File: templates/consumer/go/cmd/invoke.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		Input:             input,
+		Timeout:           types.DefaultTimeout,
+		Repeated:          types.DefaultRepeated,
+		RepeatedFrequency: types.DefaultFrequency,
+		RepeatedTotal:     types.DefaultTotal,
+	}
+
+	configPath := common.ConfigPath
+
+	if len(args) == 4 || len(args) == 5 || len(args) == 6 || len(args) == 7 || len(args) == 8 {
```
Comment (chengwenxi) [inline templates/consumer/go/cmd/invoke.go]: why not ` len(args) >= 4 && len(args)<=8`?

File: templates/consumer/go/cmd/invoke.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+		invokeConfig.Timeout, err = strconv.ParseInt(args[3], 10, 64)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	if len(args) == 5 || len(args) == 6 || len(args) == 7 || len(args) == 8 {
```
Comment (chengwenxi) [inline templates/consumer/go/cmd/invoke.go]: ditto.

File: templates/consumer/go/cmd/invoke.go [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+		invokeConfig.Repeated, err = strconv.ParseBool(args[4])
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	if len(args) == 6 || len(args) == 7 || len(args) == 8 {
```
Comment (chengwenxi) [inline templates/consumer/go/cmd/invoke.go]: ditto.

File: templates/consumer/go/cmd/invoke.go [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+	sdkTypes "github.com/irisnet/service-sdk-go/types"
+	"github.com/spf13/cobra"
+)
+
+func invokeCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "invoke",
+		Short:   "Invoke service",
+		Example: `{{service_name}}-sc invoke [provider-list] [fee-cap] [input] [timeout] [repeated] [frequency] [total] [config-file]`,
+		Args:    cobra.RangeArgs(3, 8),
```
Comment (chengwenxi) [inline templates/consumer/go/cmd/invoke.go]: Too many arguments, consider using flags for optional parameters.

File: templates/consumer/go/app/app.go [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+) App {
+	return App{
+		ServiceClient:    serviceClient,
+		ResponseCallback: {{service_name}}.ResponseCallback,
+		Logger:           logger,
+	}
+}
+
+// Start starts the provider process
+func (app App) Start(addr string, responseCallback types.ResponseCallback) {
```
Comment (chengwenxi) [inline templates/consumer/go/app/app.go]: no need the start function, move subscribe to Invoke.

File: templates/provider/go/app/app.go [hunk 8]
```diff
@@ -30,11 +33,21 @@ func NewApp(
...
-		app.Logger.Errorf("failed to register service request listener, err: %s", err.Error())
+		app.Logger.Errorf("failed to subscribe service request listener, err: %s", err.Error())
 		return
 	}
 
 	select {}
 }
+
+// Bind provider
+func (app App) Bind(bindConfig servicesdk.BindServiceRequest) {
```
Comment (chengwenxi) [inline templates/provider/go/app/app.go]: why need binding service ?

File: templates/provider/go/{{service_name}}/request_callback.go [hunk 9]
```diff
@@ -17,9 +20,8 @@ func RequestCallback(reqID, input string) (output *types.ServiceOutput, requestR
 		}
 		return nil, requestResult
 	}
-
-	_ = serviceInput
 	// Supplementary service logic...
+	_ = serviceInput
```
Comment (chengwenxi) [inline templates/provider/go/{{service_name}}/request_callback.go]: useless code

File: templates/consumer/go/app/app.go (hunk 7 above)
Comment (GTLiSunnyi) [reply templates/consumer/go/app/app.go]: Let's keep it that way for the time being.

File: templates/provider/go/app/app.go (hunk 8 above)
Comment (GTLiSunnyi) [reply templates/provider/go/app/app.go]: Let's keep it that way for the time being.

**Review by chengwenxi — APPROVED** [review]

**Pull request merged**